test/cppsuite/tests/cursor_bound_03.cpp

```diff
@@ -0,0 +1,423 @@
+         * prefix enabled is then validated using the search_near call without prefix enabled.
+         */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        const char *key_prefix;
+        int exact_prefix, ret;
+        int64_t prefix_size;
+        std::map<uint64_t, scoped_cursor> cursors;
+        std::string generated_prefix, key_prefix_str;
+
+        while (tc->running()) {
+            /* Get a random collection to work on. */
+            collection &coll = tc->db.get_random_collection();
+
+            /* Find a cached cursor or create one if none exists. */
+            if (cursors.find(coll.id) == cursors.end())
+                cursors.emplace(coll.id, std::move(tc->session.open_scoped_cursor(coll.name)));
+
+            auto &cursor_prefix = cursors[coll.id];
+
+            wt_timestamp_t ts = tc->tsm->get_valid_read_ts();
+            /*
+             * The oldest timestamp might move ahead and the reading timestamp might become invalid.
+             * To tackle this issue, we round the timestamp to the oldest timestamp value.
+             */
+            tc->txn.begin(
+              "roundup_timestamps=(read=true),read_timestamp=" + tc->tsm->decimal_to_hex(ts));
+
+            while (tc->txn.active() && tc->running()) {
+                /*
+                 * Generate a random prefix. For this, we start by generating a random size and then
+                 * its value.
+                 */
+                prefix_size = random_generator::instance().generate_integer(
+                  static_cast<int64_t>(1), tc->key_size);
+                generated_prefix = random_generator::instance().generate_random_string(
+                  prefix_size, characters_type::ALPHABET);
+
+                /* Call search near with the prefix cursor. */
+                bound_set prefix_bound = bound_set(generated_prefix.c_str());
+                prefix_bound.apply(cursor_prefix);
+                cursor_prefix->set_key(cursor_prefix.get(), generated_prefix.c_str());
+                ret = cursor_prefix->search_near(cursor_prefix.get(), &exact_prefix);
+                testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+                if (ret == 0) {
+                    testutil_check(cursor_prefix->get_key(cursor_prefix.get(), &key_prefix));
+                    key_prefix_str = key_prefix;
+                } else {
+                    key_prefix_str = "";
+                }
+
+                /* Open a cursor with the default configuration on the selected collection. */
+                scoped_cursor cursor_default(tc->session.open_scoped_cursor(coll.name));
+
+                /* Verify the prefix search_near output using the default cursor. */
+                validate_prefix_search_near(
+                  ret, exact_prefix, key_prefix_str, cursor_default, generated_prefix);
+
+                tc->txn.add_op();
```

Comment:
I don't think we need this line no more as we want to roll back.

Reply:
I'm gonna leave it as is for now.